Makefile

```
@@ -1,23 +1,66 @@
+# Go file length check (staged files only)
+check-file-length:
+	@git diff --cached --name-only --diff-filter=ACMR | grep '\.go$$' | while read f; do \
+	  if [ -f "$$f" ] && [ "$$(wc -l < "$$f")" -gt 300 ]; then \
+	    echo "FAIL: $$f has $$(wc -l < "$$f") lines (max 300)"; \
+	    exit 1; \
+	  fi; \
+	done
+	@echo "  Staged Go files under 300 lines ✓"
+
+# Go linting
+lint-go-cognitive:
```

checkmake (0.3.2)

[warning] Target body for "lint-go-cognitive" exceeds allowed length of 5 lines (7).

(maxbodylength)